src/components/icon/icon-registry.ts

```diff
@@ -0,0 +1,239 @@
+import {Injectable} from 'angular2/core';
+import {Http} from 'angular2/http';
+import {Observable} from 'rxjs/Rx';
+
+/**
+  *  Configuration for an icon, including the URL and possibly the cached SVG element.
+  */
+class IconConfig {
```

Comment:
If this class is meant to be internal (i.e., users of the component never interactive with it), add `@internal` to the description.